.github/workflows/test.yml

```diff
@@ -134,93 +138,42 @@ jobs:
-            env GOOS=windows GOARCH=arm go build -v ./...
-            env GOOS=windows GOARCH=arm64 go build -v ./...
-
-            # Check cross-compiling macOS binaries.
-            env GOOS=darwin GOARCH=amd64 go build -v ./...
-            env GOOS=darwin GOARCH=arm64 go build -v ./...
-
-            # Check cross-compiling Linux binaries.
-            env GOOS=linux GOARCH=amd64 go build -v ./...
-            env GOOS=linux GOARCH=arm64 go build -v ./...
-
-            # Check cross-compiling FreeBSD binaries.
-            env GOOS=freebsd GOARCH=amd64 go build -gcflags="github.com/ebitengine/purego/internal/fakecgo=-std" -v ./...
-            env GOOS=freebsd GOARCH=arm64 go build -gcflags="github.com/ebitengine/purego/internal/fakecgo=-std" -v ./...
-
-            echo "=> go build (plugin)"
-            # Make sure that plugin buildmode works since we save the R15 register (#254)
-            go build -buildmode=plugin ./examples/libc
-
-            echo "=> go mod vendor"
-            mkdir /tmp/vendoring
-            cd /tmp/vendoring
-            go mod init foo
-            echo 'package main' > main.go
-            echo 'import (' >> main.go
-            echo '  _ "github.com/ebitengine/purego"' >> main.go
-            echo ')' >> main.go
-            echo 'func main() {}' >> main.go
-            go mod edit -replace github.com/ebitengine/purego=$GITHUB_WORKSPACE
-            go mod tidy
-            go mod vendor
-            go build -v .
-
-            cd $GITHUB_WORKSPACE
-            echo "=> go test CGO_ENABLED=0"
-            env CGO_ENABLED=0 go test -gcflags="github.com/ebitengine/purego/internal/fakecgo=-std" -shuffle=on -v -count=10 ./...
-
-            echo "=> go test CGO_ENABLED=1"
-            env CGO_ENABLED=1 go test -shuffle=on -v -count=10 ./...
-
-            echo "=> go test CGO_ENABLED=0 w/o optimization"
-            env CGO_ENABLED=0 go test "-gcflags=all=-N -l" -v ./...
-
-            echo "=> go test CGO_ENABLED=1 w/o optimization"
-            env CGO_ENABLED=1 go test "-gcflags=all=-N -l" -v ./...
-
-            if [ -z "$(echo ${{matrix.go}} | grep '^1.1')" ]; then
-              echo "=> go test race"
-              go test -race -shuffle=on -v -count=10 ./...
-            fi
+            
+            chmod +x .github/scripts/bsd_tests.sh
+          run: bash .github/scripts/bsd_tests.sh
+      - name: Run in netbsd
+        if: matrix.os == 'NetBSD'
+        uses: vmactions/netbsd-vm@v1
+        with:
+          usesh: true
+          prepare: |
+            /usr/sbin/pkg_add -v bash
+            ftp https://go.dev/dl/go${{matrix.go}}.netbsd-amd64.tar.gz
+            mkdir /usr/local
+            rm -fr /usr/local/go && tar -C /usr/local -xf go${{matrix.go}}.netbsd-amd64.tar.gz
+            ln -s /usr/local/go/bin/go /usr/local/bin
+            export PATH=$PATH:/usr/local/go/bin/
+            
+            chmod +x .github/scripts/bsd_tests.sh
+          run: bash .github/scripts/bsd_tests.sh
```

Comment:
Isn't this already in a vm? In this case, shouldn't we copy the bsd_test.sh into the vm?

Reply:
The repo is checked out which includes the bsd_tests.sh file. There isn't a need to copy it I don't think

Reply:
https://github.com/ebitengine/purego/actions/runs/13739026602/job/38426088637#step:4:1323

I don't think this result indicates the file existence...

Reply:
I see the .github folder though in that ls call

Reply:
Ok it looks like it's not there at least in FreeBSD

```
.
  ..
  .cshrc
  .k5login
  .login
  .profile
  .sh_history
  .shrc
  .ssh
  go1.20.14.freebsd-amd64.tar.gz
  work
  
  ./.ssh:
  .
  ..
  authorized_keys
  config
  id_rsa
  id_rsa.pub
  known_hosts
  chmod: .github/scripts/bsd_tests.sh: No such file or directory
```

Reply:
I'm not sure but what about `$GITHUB_WORKSPACE/.github/scripts/bsd_tests.sh`?

Reply:
closer now since some of the NetBSD ones passed. FreeBSD still failed tho

Reply:
Interesting, `$GITHUB_WORKSPACE` seems empty now...